src/otx/v2/adapters/torch/dataset.py

```diff
@@ -0,0 +1,210 @@
+            batch_size (int): The batch size to use.
+            num_workers (int): The number of worker processes to use for data loading.
+            shuffle (bool): Whether to shuffle the data.
+            pin_memory (bool): Whether to pin memory for faster GPU transfer.
+            drop_last (bool): Whether to drop the last incomplete batch.
+            sampler (Optional[Union[Sampler, Iterable, Dict]]): The sampler to use for data loading.
+            persistent_workers (bool): Whether to keep the worker processes alive between iterations.
+            **kwargs: Additional arguments to pass to the DataLoader constructor.
+
+        Returns:
+            Optional[TorchDataLoader]: The DataLoader for the given dataset.
+        """
+        if dataset is None:
+            return None
+
+        # Sampler
+        if isinstance(sampler, dict):
+            pass
+        if sampler is not None:
+            shuffle = False
+
+        dataloader = TorchDataLoader(
+            dataset,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            shuffle=shuffle,
+            drop_last=drop_last,
+            persistent_workers=persistent_workers,
+            **kwargs,
+        )
+        sampler_cfg: dict | list | None = None
+        if isinstance(sampler, dict):
+            sampler_cfg = sampler
+        elif isinstance(sampler, Sampler):
+            sampler_class_name = getattr(sampler.__class__, '__qualname__', None)
+            sampler_cfg = {"type": sampler_class_name} if sampler_class_name else {"type": str(sampler.__class__)}
+        elif isinstance(sampler, Iterable):
+            sampler_cfg = []
+            for s in sampler:
+                sampler_class_name = getattr(s.__class__, '__qualname__', None)
+                sampler_cfg.append({"type": sampler_class_name} if sampler_class_name else {"type": str(s.__class__)})
+
+        dataset_cfg = dataset.configs if hasattr(dataset, "configs") else dataset
+        dataloader.configs = {
+            "batch_size": batch_size,
+            "sampler": sampler_cfg,
+            "num_workers": num_workers,
+            "pin_memory": pin_memory,
+            "shuffle": shuffle,
+            "persistent_workers": persistent_workers,
+            "dataset": dataset_cfg,
+            **kwargs,
+        }
+        return dataloader
+
+    def subset_dataloader(
+        self,
+        subset: str,
+        pipeline: dict | list | None = None,
+        batch_size: int | None = None,
+        num_workers: int | None = None,
+        config: str | dict | None = None,
+        shuffle: bool = True,
+        pin_memory: bool = False,
+        drop_last: bool = False,
+        sampler: Sampler | (Iterable | dict) | None = None,
+        persistent_workers: bool = False,
+        **kwargs,
+    ) -> TorchDataLoader:
+        r"""Torch Based Dataset.subset_dataloader.
```

Comment:
Can you add a bit more description? Without looking at the code, the description is not sufficient to understand what it does.

Reply:
I've added a more specific description for this.